src/rhsmlib/facts/cloud_facts.py

```diff
@@ -262,6 +276,101 @@ def get_azure_facts(self):
                     facts['azure_offer'] = values['compute']['offer']
         return facts
 
+    def is_gcp(self):
+        """
+        Is VM running on GCP public cloud
+        :return: True, when the system is running on GCP. Otherwise, return False.
+        """
+        hw_info = self._collected_hw_info
+
+        if 'dmi.bios.vendor' in hw_info and \
```

Comment:
Is it possible for the dmi.bio.vendor value to be None here?

Reply:
Good point. The `hw_info` could be `None`.